weighted_selection.py

```diff
@@ -0,0 +1,20 @@
+import random
+
+def weighted_random_selection(weights):
+    total_weight = sum(weights)
+    probabilities = [item / total_weight for item in weights]
+    cumulative_probabilities = [sum(probabilities[:i+1]) for i in range(len(probabilities))]
```

Comment:
It is a very CPU-expensive statement for large lists. Do you know why?

Reply:
yes, for large data it is expensive as summing large datasets can be costly due to the repeated iterations involved in the process.

Reply:
So, improve it.